typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -504,6 +519,17 @@ export class WarpCore {
     const amountError = this.validateAmount(originTokenAmount);
     if (amountError) return amountError;
 
+    const destinationCollateralError = await this.validateDestinationCollateral(
```

Comment:
I'm open to validating collateral here, probably a good idea, but note we'll want to then remove the `isDestinationCollateralSufficient` call in useTokenTransfer in the warp UI to avoid running this twice.

Reply:
Seems reasonable to me. 

IMO best to do all the validations we can before entering the transaction building flow and updating the UI to make it look like this will tx will work.

Reply:
Also nice here because then the CLI benefits from these checks without changes